### DISCUSSION TITLE
前端应该如何开发调试

### DISCUSSION THREAD

**Discussion by Linya-IronMan** [AlistGo/alist#4431] [Q&A]
- 已有一个最新版本的 alist 运行在 Linux 上，web 可正常访问 http://192.168.2.22:5244
- 想要添加一些功能，所以 git clone alist-web 项目最新版到本地
- 参照教程，执行 pnpm i 安装相关依赖成功
- 参照教程，执行 npm run build 失败
- 错误如下，找不到 solid-router/src 目录
![image](https://github.com/alist-org/alist/assets/77786952/cb59cfdb-2dca-4a59-9340-85ba19e0da9e)

- 找到一个相关的问题描述
https://github.com/alist-org/alist/issues/1763
疑似 3.10.1 以上版本的问题 
- git clone alist-web 3.10.0 版本，安装依赖并构建项目
出现同样的问题

【问题：最新版本的 alist-web 应该如何构建？】

---

- 觉得每次都要构建项目，挪动文件，然后才能查看修改效果，过于麻烦
- 尝试无需构建就能查看 web 效果

- 环境
- 有一个 Ubuntu 运行最新版本 alist 192.168.2.22:5244
- 计划在本地运行 alist-web，将后端接口指向 Ubuntu 启动的 alist
- 修改了项目中的后端指向地址 localhost => 192.168.2.22 
- 【vite.config.ts, .env.development, src/store/local_settings.ts】
![image](https://github.com/alist-org/alist/assets/77786952/633fdd5d-df6f-4ad7-ac9f-4a251d9a0e8a)
![image](https://github.com/alist-org/alist/assets/77786952/66819077-8c61-4d2a-8b7c-6b22beacf340)
![image](https://github.com/alist-org/alist/assets/77786952/68792787-ba52-4ba2-8f34-e019c7afc9d2)

- 之后执行 npm start 
![image](https://github.com/alist-org/alist/assets/77786952/3ac19c85-9bc9-41fd-bfa6-c2ba73878767)

【问题：应该如何搭建开发调试环境？】

**Comment by anwen-anyi** [comment]
手把手教程🙄
https://anwen-anyi.github.io/index/build/kaifa.html

**Reply by Linya-IronMan** [reply]
...... 谢谢，一条条命令对比下来，找到了问题。
clone alist-web 的时候 有添加 --recurse-submodules ，这样才会有 solid-router 中的文件

**Reply by anwen-anyi** [reply]
文档写的这个也可以啊 你没仔细看吧qaq
`git clone --recursive https://github.com/alist-org/alist-web.git`
